### PR TITLE
Fix *Env classes

### PR DESCRIPTION
**Pull request by be-marc**
* Removes setting of `cols` and `rows`
* Add `features` to parameter set

### CODE REVIEW

File: R/ResamplingSpCVBuffer.R [hunk 1]
```diff
@@ -5,6 +5,17 @@
...
+#' @note
+#' The `blockcv::buffering` function was written for species distribution modeling with presence-absence data.
+#' However, the default settings allow to conduct a leave-one-out cross validation
+#' for two-class, multi-class and continuous response data, where each observation is one test set.
+#' For each test, all observations outside the buffer around the test observation are included in the training set.
+#'
+#' The parameter `spDataType = PB` and `addBG` are designed for presence-background data in species distribution modelling.
+#' If `spDataType = PB`, test sets are only created for each presence observation (`task\$postive`).
+#' The option `addBG = TRUE` adds the background data inside the buffer to the corresponding test sets.
+#' For each test set, all observations outside the buffer around the test observation are included in the training set.
```
Comment (pat-s) [inline R/ResamplingSpCVBuffer.R]: Please format to a width of 80. Can be done easily by highlighting the desired text and then hitting `ctrl + Shift + /` in RStudio.

File: R/ResamplingSpCVBuffer.R [hunk 2]
```diff
@@ -34,9 +45,11 @@ ResamplingSpCVBuffer = R6Class("ResamplingSpCVBuffer",
     #'   Identifier for the resampling strategy.
     initialize = function(id = "spcv-buffer") {
       ps = ParamSet$new(params = list(
-        ParamInt$new("range", lower = 1L, tags = "required")
+        ParamInt$new("theRange", lower = 1L, tags = "required"),
+        ParamFct$new("spDataType", default = "PA", levels = c("PA", "PB")),
+        ParamLgl$new("addBG", default = TRUE)
       ))
-      ps$values = list(range = 100)
```
Comment (pat-s) [inline R/ResamplingSpCVBuffer.R]: Oh yeah, we should not set default from our side. Good catch!

File: R/ResamplingSpCVBuffer.R [hunk 3]
```diff
@@ -72,47 +89,55 @@ ResamplingSpCVBuffer = R6Class("ResamplingSpCVBuffer",
...
-    .sample = function(ids, coords, crs) {
+    .sample = function(ids, response, coords, positive, crs, properties) {
 
       require_namespaces(c("blockCV", "sf"))
 
-      points = sf::st_as_sf(coords,
+      pars = self$param_set$get_values()
+
+      if (!isTRUE("twoclass" %in% properties) && isTRUE(pars$spDataType == "PB")) {
+        stop("spDataType = 'PB' should only be used with two-class response")
```
Comment (pat-s) [inline R/ResamplingSpCVBuffer.R]: Please use `mlr3misc::stopf()`. Comes also with a nice option `wrap` that you might want to check out.

File: R/ResamplingSpCVBuffer.R [hunk 4]
```diff
@@ -72,47 +89,55 @@ ResamplingSpCVBuffer = R6Class("ResamplingSpCVBuffer",
...
+      }
+
+      if (!is.null(pars$addBG) && isTRUE(pars$spDataType == "PA")) {
+        stop("Parameter addBG should only be used with spDataType = 'PB'")
+      }
+
+      # Recode response to 0/1 for twoclass
+      if (isTRUE("twoclass" %in% properties)) {
```
Comment (pat-s) [inline R/ResamplingSpCVBuffer.R]: `isTRUE()` is problematic due to differing behavior in R > 3.5. Can be simply omitted here.

File: R/ResamplingSpCVBuffer.R [hunk 5]
```diff
@@ -72,47 +89,55 @@ ResamplingSpCVBuffer = R6Class("ResamplingSpCVBuffer",
...
-        theRange = self$param_set$values$range,
-        progress = FALSE
-      )
+      inds = invoke(blockCV::buffering,
+        speciesData = data,
+        progress = FALSE,
+        .args = pars)
 
-      inds = map(inds$folds, function(x) {
+      map(inds$folds, function(x) {
```
Comment (pat-s) [inline R/ResamplingSpCVBuffer.R]: We should probably prefix `map` & friends with `mlr3misc::` to be more explicit here.

File: R/autoplot.R [hunk 6]
```diff
@@ -123,11 +123,91 @@ autoplot.ResamplingSpCVBuffer = function(
...
+  coords$row_id = task$row_ids
+  require_namespaces(c("sf", "cowplot"))
+
+  # instantiate if not yet done
+  if (!resampling$is_instantiated) {
+    resampling = resampling$instantiate(task)
+  }
+
+  if (is.null(fold_id)) {
+    stop("Plotting all folds of a LOOCV instance is not supported. Please provide a fold ID.") # nolint
```
Comment (pat-s) [inline R/autoplot.R]: Good case for `stopf(wrap = TRUE)`. (I guess this was old code from myself).

File: tests/testthat/helper.R [hunk 7]
```diff
@@ -10,3 +10,53 @@ task_grp = TaskClassifST$new("ecuador-grp", as_data_backend(data),
   target = "slides",
   coordinate_names = c("x", "y"))
 task_grp$col_roles$group = "grp"
+
+# Create 6x6 point grid with 1m distance between points
+TEST_MAKE_SP = function() {
```
Comment (pat-s) [inline tests/testthat/helper.R]: Why all upper case?

File: tests/testthat/test_spcv_buffer.R [hunk 8]
```diff
@@ -18,6 +18,8 @@ test_that("grouping throws errors when 'groups' and 'stratify' is set", {
 
 test_that("train and test set getter functions are working", {
   spcv_rsp_buffer = mlr_resamplings$get("spcv-buffer")
+  spcv_rsp_buffer$param_set$values = insert_named(spcv_rsp_buffer$param_set$values,
+    list(theRange = 1))
```
Comment (pat-s) [inline tests/testthat/test_spcv_buffer.R]: does the user also need to use `insert_named()`?

File: tests/testthat/helper.R (hunk 7 above)
Comment (be-marc) [reply tests/testthat/helper.R]: We are doing this in `mlr3tuning` for functions that create test objects (https://github.com/mlr-org/mlr3tuning/blob/master/tests/testthat/helper.R). However, we are not doing this mlr3.

File: R/ResamplingSpCVBuffer.R (hunk 1 above)
Comment (be-marc) [reply R/ResamplingSpCVBuffer.R]: This makes my life easier

File: tests/testthat/test_spcv_buffer.R (hunk 8 above)
Comment (be-marc) [reply tests/testthat/test_spcv_buffer.R]: Only if the user does not want to overwrite his old settings of other parameters.

**Comment by be-marc** [comment]
@pat-s Okay I integrated your suggestions. Might be ready for merging now.

**Pull request merged**